src/utils/resources/shared.ts

```diff
@@ -339,6 +338,14 @@ export const getNamespace = <A extends K8sResourceCommon = K8sResourceCommon>(re
 export const getUID = <A extends K8sResourceCommon = K8sResourceCommon>(resource: A): string =>
   resource?.metadata?.uid;
 
+/**
+ *
+ * @param resource k8s resource with optional cluster
+ * @returns resource's cluster
+ */
+export const getCluster = <A extends K8sResourceCommon = K8sResourceCommon>(resource?: A) =>
```

Comment:
I would move this to the multicluster directory since it's only relevant in a multicluster environment.